codes/quantum/properties/symmetry_protected.yml

```diff
@@ -0,0 +1,31 @@
+code_id: symmetry_protected
+
+name: 'Symmetry-protected self-correcting quantum code'
+introduced: '\cite{arXiv:1805.01474}'
+
+description: |
+  The physical degrees of freedom of the code correspond to a physical lattice with finite-dimensional local Hilbert space. The codespace is the degenerate ground space of a Hamiltonian that consists of terms localized in space. The system is required to have a symmetry not spontaneously broken, with the symmetry-protected self-correcting property described as follows.
+
+  Suppose the physical state is \(\rho(0)\) at time \(t=0\), and is evolved to \(\rho(t)\) by the Hamiltonian plus a small coupling to a thermal bath with a low enough temperature, in the way that is local on the lattice and respects the symmetry. Then the memory time \(\tau_\text{mem}\) is defined to be the maximal \(t\) such that \(\left\|\Phi_\text{ec}(\rho(t))-\rho(0)\right\|_1\leq\epsilon\), where \(\Phi_\text{ec}\) is the error correcting map and \(\epsilon\) a fixed error rate.
+  \(\tau_\text{mem}\) is required to grow without bound (typically exponentially) in the system size.
+  In addition, the logical operators are required to have a decomposition into a product such that each factor is localized in space and has the symmetry.
+
+  Such a code can be constructed using a 3D symmetry-protected topological (SPT) phase in the bulk with a 1-form symmetry, and an anomalous 2D symmetry-enriched topological (SET) phase on the boundary with ground-state degeneracy, such that the anyons on the boundary are confined by the string-like excitations in the bulk enforced by the symmetry.
```

Comment:
Careful with audience here; "anyons" "SPT" "SET" "string-like" "confined" all are not defined. It's ambitious to cover all that here. I suggest elaborating that on-site (i.e., tensor-product with weight equal to number of subsystems) is not sufficient, and that 1-form (define...) is.